py/plugins/google-cloud/src/genkit/plugins/google_cloud/telemetry/tracing.py

```diff
@@ -0,0 +1,162 @@
+        client=None,
+        resource_regex=None,
+    ):
+        """Initializes the GenkitGCPExporter.
+
+        Args:
+            project_id: GCP project ID for the project to send spans to. Alternatively, can be
+                configured with :envvar:`OTEL_EXPORTER_GCP_TRACE_PROJECT_ID`.
+            client: Cloud Trace client. If not given, will be taken from gcloud
+                default credentials
+            resource_regex: Resource attributes with keys matching this regex will be added to
+                exported spans as labels (default: None). Alternatively, can be configured with
+                :envvar:`OTEL_EXPORTER_GCP_TRACE_RESOURCE_REGEX`.
+        """
+        super().__init__(
+            project_id=project_id,
+            client=client,
+            resource_regex=resource_regex,
+        )
+
+    def export(self, spans: Sequence[ReadableSpan]) -> SpanExportResult:
+        """Export the spans to Cloud Trace.
+
+        Iterates through the provided spans, and adds an attribute to the spans.
+
+        Note:
+            Leverages span transformation and formatting to opentelemetry-exporter-gcp-trace.
+            See: https://cloud.google.com/python/docs/reference/cloudtrace/latest
+
+        Args:
+            spans: A sequence of OpenTelemetry ReadableSpan objects to export.
+
+        Returns:
+            SpanExportResult.SUCCESS upon successful processing (does not guarantee
+            server-side success).
+        """
+        try:
+            self.client.batch_write_spans(
+                request=BatchWriteSpansRequest(
+                    name=f"projects/{self.project_id}",
+                    spans=self._translate_to_cloud_trace(spans),
+                )
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.error("Error while writing to Cloud Trace", exc_info=ex)
+            return SpanExportResult.FAILURE
+
+        return SpanExportResult.SUCCESS
+
+    def add_tracer_attributes(
+        self, spans: Sequence[ReadableSpan]
+    ) -> Sequence[ReadableSpan]:
+        """Adds the instrumentation library attribute.
+
+        Args:
+            spans: Sequence of spans to modify.
+
+        Returns:
+            Sequence of spans modified.
+        """
+        modified_spans: list[ReadableSpan] = []
+
+        for span in spans:
+            modified_spans.append(
+                span.attributes.update(
+                    {
+                        'instrumentationLibrary': {
+                            'name': 'genkit-tracer',
+                            'version': 'v1',
+                        },
+                    }
+                )
+            )
+
+        return modified_spans
+
+
+def init_telemetry_gcp_exporter() -> (SpanExporter | None):
+    """Initializes tracing with a provider and optional exporter."""
+    telemetry_project_id = os.environ.get('GOOGLE_CLOUD_PROJECT')  # TODO: set correct envvar
```

Comment:
the project ID should be automatically retrieved from the GoogleAuth client. Env var should not be required.
https://googleapis.dev/python/google-auth/latest/user-guide.html#application-default-credentials